### DISCUSSION TITLE
Is it possible to create files with curl?

### DISCUSSION THREAD

**Discussion by ghost** [curl/curl#9594] [Q&A] [answered]
Hi all.

I would like to know if it is possible to create files with curl. *Is it possible to create files with curl?*

#### example
> curl -url ./folder/file.txt

**Comment by cmeister2** [comment, marked as answer]
You can use either `--output` or `-o` to output from curl to a file.

```shell
curl --output out.txt https://www.google.com
curl -o out.txt https://www.google.com
```

**Reply by ghost** [reply]
Hi cmeister2. something like to... ``curl file:///path/to/file.txt``

**Reply by piru** [reply]
`curl --upload-file file.txt file:///path/to/file.txt`

**Reply by ghost** [reply]
Hi piru. would it be possible to have this protocol without ``file://``? something like to:
``curl ./folder/file.txt`` vs ``curl file:///path/to/file.txt`` ? Is it possible this with curl?

**Reply by bagder** [reply]
No, why would curl support that? It works with URLs.

**Reply by ghost** [reply]
Hi bagder, so thank you for feedback.

**Reply by piru** [reply]
It is not possible.

**Reply by ghost** [reply]
Hi piru. why is it not possible? is this not technically possible or maybe because curl only works with urls? (it's a curious question).

**Reply by piru** [reply]
Technically it probably is possible to add it, but I think it would add too much ambiguity to url parsing (which already is hard enough as it is).

However, you don't need to agree. If you really must have it, you can always fork curl and add it yourself.

**Reply by ghost** [reply]
Hi piru.

> Technically it probably is possible to add it, but I think it would add too much ambiguity to url parsing (which already is hard enough as it is).

My question has now been answered. thank you very much for your feedback.

**Reply by cmeister2** [reply]
> Hi piru. why is it not possible? is this not technically possible or maybe because curl only works with urls? (it's a curious question).

Not piru but - what is the expected outcome of `curl ./folder/file.txt`?

- create empty file at `./folder/file.txt`: use `touch` instead, curl isn't going to just create random files if it doesn't need to.
- show the contents of `./folder/file.txt`: it's not a URL, and curl only works with URLs.
- write something to `./folder/file.txt`: where's the data coming from?

**Reply by ghost** [reply]
Hi all. Anyway, thanks a lot for the feedback.
